core/src/main/java/net/jonathangiles/tools/maven/whitelistgenerator/report/JSONReport.java

```diff
@@ -14,17 +16,33 @@ public String getName() {
     }
 
     @Override
-    public void report(Result result, Path reportFile) {
-        try (FileWriter writer = new FileWriter(reportFile.toFile()))  {
+    public void report(Result result, Path reportFilePath) {
+        FileWriter writer = null;
+        try {
+            File reportFile = reportFilePath.toFile();
+            if (!reportFile.exists()) {
+                System.out.println("Creating file " + reportFile.getAbsolutePath());
+                reportFile.getParentFile().mkdirs();
```

Comment:
Should we put this into a conditional to handle the case where creating the directories fail?

Reply:
If the directories cannot be created, the file cannot be created too which will throw an exception and is caught in the `catch` block to log the stacktrace.